### ISSUE TITLE
inconsistency locating points switch isRegular on/off

### ISSUE THREAD

**Issue by julesghub** [underworldcode/underworld2#293]
The difference is caused by _ElementCellLayout_CellOf() and the if statement re: isRegular.
The end result is the isRegular case uses a fudge factor, whilst Mesh_Algorithms_SearchWithFullIncidence() (nont isRegular) does a basic geometric filter on the mesh without a fudge factor.

Code to recreate the issue.
```python
import underworld as uw
from underworld import function as fn
import glucifer
import numpy as np
mesh=uw.mesh.FeMesh_Cartesian()

swarm = uw.swarm.Swarm(mesh)
array = np.zeros((2,2))
print(mesh._cself.isRegular)
array[0] = [1. - 1e-13,0.5] 
array[1] = [1. + 1e-13,0.5] 
swarm.add_particles_with_coordinates(array)
print(swarm.particleGlobalCount)
with mesh.deform_mesh():
    pass
print(mesh._cself.isRegular)
swarm.add_particles_with_coordinates(array)
```

**Comment by julesghub** [comment]
The bug is the inconsistent results, not the fudge.
@rbeucher and I believe his non regular 3D model has been experiencing this issue. UpdateOwners() is using a fudge factor, but the evaluate function is using the non fudge factor algorithm.
We are currently testing and will report more later.

The fudge factor is  https://github.com/underworldcode/underworld2/blob/master/libUnderworld/StgFEM/Discretisation/src/FeMesh_ElementType.c#L106

**Comment by rcarluccio** [comment]
Hi all,

Has this issue been addressed? I am running the underworld version '2.5.0b' and I see an inconsistency in the use of the function: fn.exception.SafeMaths( fn.misc.max(x, y) ), especially when it is used in combination with x= fn.exception.SafeMaths( fn.misc.min(a, b) ).

**Comment by jmansour** [comment]
Hi Roberta

I don't think this is resolved, but I'm guessing this isn't related to your min/max inconsistencies.  

Can you post a new issue with a minimal example demonstrating your problem?
